x-pack/plugins/apm/server/saved_objects/migrations/migrate_legacy_apm_indices_to_space_aware.test.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { CoreStart, Logger } from 'src/core/server';
+import {
+  APM_INDEX_SETTINGS_SAVED_OBJECT_ID,
+  APM_INDEX_SETTINGS_SAVED_OBJECT_TYPE,
+} from '../../../common/apm_saved_object_constants';
+import { migrateLegacyAPMIndicesToSpaceAware } from './migrate_legacy_apm_indices_to_space_aware';
+
+const loggerMock = {
+  debug: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn(),
+} as unknown as Logger;
+
+describe('migrateLegacyAPMIndicesToSpaceAware', () => {
+  describe('when legacy APM indices is not found', () => {
+    const mockBulkCreate = jest.fn();
+    const mockFind = jest.fn();
+    const core = {
+      savedObjects: {
+        createInternalRepository: jest.fn().mockReturnValue({
+          get: () => {
+            throw new Error('BOOM');
+          },
+          find: mockFind,
+          bulkCreate: mockBulkCreate,
+        }),
+      },
+    } as unknown as CoreStart;
+
+    it('does not save any new saved object', () => {
+      migrateLegacyAPMIndicesToSpaceAware({
```

Comment:
Missing `await`?
```suggestion
      await migrateLegacyAPMIndicesToSpaceAware({
```

Reply:
Yeah, this operation is very quick and could be done in the background.